connect/mirror/src/test/java/org/apache/kafka/connect/mirror/integration/DedicatedMirrorIntegrationTest.java

```diff
@@ -280,7 +280,12 @@ public void testMultiNodeCluster() throws Exception {
             // Cluster aliases
             final String a = "A";
             // Use a convoluted cluster name to ensure URL encoding/decoding works
-            final String b = "B- ._~:/?#[]@!$&'()*+;=\"<>%{}|\\^`618";
+            // TODO: Jetty 12 throws a 400 amiguous error with the previous test string here
+            // It seems like at least some encoded characters are not allowed anymore based on
+            // https://github.com/jetty/jetty.project/issues/11890#issuecomment-2156442947
+            // See https://github.com/jetty/jetty.project/issues/11890
+            //final String b = "B- ._~:/?#[]@!$&'()*+;=\"<>%{}|\\^`618";
+            final String b = "B";
```

Comment:
Could you keep most of these suspicious characters, and just remove the ones which are ambiguous?
From reading the linked issue, it might just be the `/` character, and the remainder are still valid.

It looks like this test was sensitive enough to find the underlying change in the library, so I'd like to keep it sensitive in case something like this happens again.

Reply:
So there's actually a few characters that are not allowed, control characters, backslash, etc. I went ahead and removed a few problem characters so it works now. I also updated the comments for the section. You can see the exact rules in the `Rejecting Suspicious Sequences` section as part of https://jakarta.ee/specifications/servlet/6.0/jakarta-servlet-spec-6.0#uri-path-canonicalization